R/mock-session.R

```diff
@@ -75,41 +75,149 @@ mapNames <- function(func, vals) {
+#' error.
+#' @include shiny.R
+#' @noRd
+makeError <- function(name, msg = paste0(name, " is for internal use only.")) {
+  if (!(name %in% names(ShinySession$public_methods)))
+    stop(name, " is not public method of ShinySession.")
+  impl <- ShinySession$public_methods[[name]]
+  body(impl) <- rlang::expr({
+    base::stop(!!msg)
+  })
+  impl
+}
+
+#' Accepts a series of named arguments. Each name corresponds to a ShinySession
+#' public method that should signal an error, and each argument corresponds to
+#' an error message.
+#' @noRd
+makeErrors <- function(...) {
+  errors <- rlang::list2(...)
+  mapply(makeError, names(errors), errors, USE.NAMES = TRUE, SIMPLIFY = FALSE)
+}
+
+#' @noRd
+makeExtraMethods <- function() {
+  c(makeWarnNoops(
+    "allowReconnect",
+    "doBookmark",
+    "exportTestValues",
+    "getBookmarkExclude",
+    "getTestSnapshotUrl",
+    "incrementBusyCount",
+    "onBookmark",
+    "onBookmarked",
+    "onRestore",
+    "onRestored",
+    "reactlog",
+    "registerDataObj",
+    "reload",
+    "resetBrush",
+    "sendBinaryMessage",
+    "sendCustomMessage",
+    "sendInputMessage",
+    "sendInsertTab",
+    "sendInsertUI",
+    "sendNotification",
+    "sendProgress",
+    "sendRemoveTab",
+    "sendRemoveUI",
+    "setBookmarkExclude",
+    "setShowcase",
+    "showProgress"
+  ), makeErrors(
+    #closed = "for internal use only"
+    `@uploadEnd` = "for internal use only",
+    `@uploadInit` = "for internal use only",
+    `@uploadieFinish` = "for internal use only",
+    createBookmarkObservers = "for internal use only",
+    dispatch = "for internal use only",
+    handleRequest = "for internal use only",
+    requestFlush = "for internal use only",
+    saveFileUrl = "for internal use only",
+    startTiming = "for internal use only",
+    wsClosed = "for internal use only"
+  ))
+}
+
+#' @description Adds generated instance methods to a MockShinySession instance.
+#'   Note that `lock_objects = FALSE` must be set in the call to `R6Class()`
+#'   that produced the generator object of the instance.
+#' @param instance instance of an R6 object, generally a `MockShinySession`.
+#' @param methods named list of method names to method implementation functions.
+#'   In our typical usage, each function is derived from a public method of
+#'   `ShinySession`. The environment of each implementation function is set to
+#'   `instance$.__enclos_env` before the method is added.
+#' @noRd
+addGeneratedInstanceMethods <- function(instance, methods = makeExtraMethods()) {
+  mapply(function(name, impl) {
+    environment(impl) <- instance$.__enclos_env__
+    instance[[name]] <- impl
+  }, names(methods), methods)
+}
+
+#' @export
+makeMockSession <- function() {
+  MockShinySession$new()
+}
+
+
 #' Mock Shiny Session
 #'
 #' @description
 #' An R6 class suitable for testing that simulates the `session` parameter
 #' provided to Shiny server functions or modules.
 #'
-#' @include timer.R
-#' @export
+#' @noRd
```

Comment:
Can we restore the previous state of the roxygen comments (that is, export it, and have the methods documented)?